packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -0,0 +1,301 @@
+      reject("GET_LAST_RUN_END_STATE", "Error getting Last Run End State", nil)
+    }
+  }
+  
+  @objc
+  func setReactNativeSDKVersion(_ version: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.embraceReactNativeSdkVersion.rawValue, value: version)
+      resolve(true)
+    } catch let error {
+      reject("SET_RN_SDK_VERSION", "Error setting ReactNative SDK version", error)
+    }
+  }
+  
+  
+  @objc
+  func setUserName(_ userName: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userName = userName
+    resolve(true)
+  }
+  
+  @objc
+  func clearUserEmail(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userEmail = nil
+    resolve(true)
+  }
+  
+  @objc
+  func setJavaScriptPatchNumber(_ patch: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.javaScriptPatchNumber.rawValue, value: patch)
+      resolve(true)
+    } catch let error {
+      reject("SET_JAVASCRIPT_PATCH_NUMBER", "Error setting JavasScript Patch Number", error)
+    }
+  }
+  
+  @objc
+  func clearUserIdentifier(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userIdentifier = nil
+    resolve(true)
+  }
+  
+  @objc
+  func clearUsername(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userName = nil
+    resolve(true)
+  }
+  
+  
+  @objc
+  func endSession(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.endCurrentSession()
+    resolve(true)
+  }
+  
+  @objc
+  func checkAndSetCodePushBundleURL(
+    _ resolve: @escaping RCTPromiseResolveBlock,
+    rejecter reject: @escaping RCTPromiseRejectBlock
+  ) {
+    do {
+#if canImport(CodePush)
+      guard let url = try CodePush.bundleURL() else {
+        reject("GET_CODEPUSH_BUNDLE_URL", "Error getting Codepush Bundle URL", nil)
+        return
+      }
+      
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.javaScriptBundleURL.rawValue, value: url.path)
+      resolve(true)
+      
+#else
+      resolve(false)
+#endif
+    } catch let error {
+      reject("CHECK_AND_SET_CODEPUSH_BUNDLE_URL", "Error setting Codepush Bundle URL", error)
+    }
+  }
+  
+  @objc(addUserPersona:resolver:rejecter:)
+  func addUserPersona(_ persona: String, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.add(persona: PersonaTag(persona), lifespan: .session)
```

Comment:
is the casting of `PersonaTag` needed here? From the iOS docs seems like you could just pass a string: https://embrace.io/docs/ios/open-source/metadata-personas/#adding-user-personas-in-the-embrace-sdk